### PR TITLE
added the new endpoint and also cleaned up the codebase

### PR DESCRIPTION
**Pull request by nfegard**
It works in Postman  ¯\_(ツ)_/¯

### CODE REVIEW

**Comment by tforrest** [comment]
Can I see the results from Postman on here? @nfegard

File: service.go [hunk 1]
```diff
@@ -259,11 +216,178 @@ func RejectFriendRequest(ctx context.Context, tokenObj *token.Token, friendAddRe
...
+	} else if err != nil {
+		e := &juniper.Error{
+			Description: "Datastore error",
+			Code:        "temp"}
+		return juniper.Response{
+			HTTPStatus: 500,
+			Err:        e,
+			Version:    version}
+	}
+	return juniper.Response{}
```
Comment (tforrest) [inline service.go]: Should return nil

File: service.go [hunk 2]
```diff
@@ -294,29 +418,19 @@ func friendRequestExists(ctx context.Context, initiator, recipient *datastore.Ke
...
+func friendAdd(ctx context.Context, queryField string, tokenObj *token.Token, requestSchema FriendAddRequest) juniper.Response {
 	requestorUsername := tokenObj.UserIdentifier
 
 	Initiator, err := findUserKeyBy(ctx, "Username", requestorUsername)
-	if err == datastore.Done {
-		return 404, "You are not found"
-	} else if err != nil {
-		return 500, "Datastore error"
+	resp := validateUser(err)
+	if resp != (juniper.Response{}) {
```
Comment (tforrest) [inline service.go]: `resp != nil`

Comment (nfegard) [inline service.go]: We can't have juniper.Response{} be replated by nil because resp can never be nil since it is a struct

File: service.go (hunk 1 above)
Comment (tforrest) [inline service.go]: oooohoho

**Comment by tforrest** [comment]
+11

**Pull request merged**